File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set=new HashSet<Integer>();
        for(int num:nums) set.add(num);
        int count=0;
        //If we traverse in org array we get time limit exceeded becoz there may be many duplicate values like 0 0 every time we traverse it get executed again and again it will exceed the time limit so we traverse through set where duplicate values neglected
        for(int num:set){
            if(!set.contains(num-1)){
                int currcount=0;
                int j=num;
                while(set.contains(j)){
                    j++;
                    currcount++;
                }
                count=Math.max(count,currcount);
            }
        }
        return count;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset=set(nums)
        count=0
        for num in numset:
            if num-1 not in numset:
                current=num
                curcount=0
                while current in numset:
                    current+=1
                    curcount+=1
                count=max(count,curcount)
        return count

        